blog/_posts/2016-10-06-julia-0.5-highlights.md

```diff
@@ -0,0 +1,726 @@
+
+- The `@view` macro was introduced, allowing the use of natural slicing syntax for array views.
+In other words you can write `@view A[1:end-1,2:end]` instead of `view(A, 1:size(A,1)-1, 2:size(A,2))`.
+
+Since views are an such important tool for both performance and for expressing complex mutating operations on arrays, especially with higher order functions, which are now fast, we may introduce a special syntax for view slices in the future.
+In particular, the syntax `A@[I...]` had a fair amount of popular support.
+Stay tuned!
+
+## And more...
+
+This is far from the full extent of the improvements introduced in Julia 0.5, but this blog post is already getting quite long, so I'll just summarize a few of the other big ticket items:
+
+- The set of string types and operations has been significantly simplified and streamlined.
+The `ASCIIString` and `UTF8String` types have been merged into a single `String` type, and the `UTF16String` and `UTF32String` and other types and functions related to string encodings other than UTF-8 have been moved into the `LegacyStrings` package, which keeps the same implementations as 0.4.
+In the future, better support for different string encodings will be developed under the `StringEncodings` package.
+
+- Most functionality related to prime generation, primality checking and combinatorics, has been moved into one of the two external packages: `Primes` or `Combinatorics`.
+To use these functions now you'll need to install the appropriate package and do `using Primes` or `using Combinatorics`.
+
+- Julia's LLVM version was upgraded from 3.3 to 3.7.1.
+This may not seem like a big deal, but the transition required herculean effort by many core Julia contributors.
+For a series of different and impossibly annoying reasons, LLVM versions 3.4 through 3.6 were not usable for Julia, so we're very happy to be back to using current versions of our favorite compiler framework.
+
+- Support for compiling and running Julia on [ARM] chips is much improved and 0.5 introduced initial support for [Power] systems, which we will be improving even further in the future.
+With support for these architectures, Julia is becoming a powerful platform mainframes and embedded computing.
+
+[ARM]: https://en.wikipedia.org/wiki/ARM_architecture
+[Power]: https://en.wikipedia.org/wiki/Power_Architecture
+
+- The 0.5 release has experimental multithreading support.
+This isn't ready for production usage, but it's fun to play around with and you can already get impressive performance gains – scalability is a key focus for Julia's multithreading.
+Note that this is true concurrent execution like C++, Go or Java – Julia is compiled so there's no [global interpreter lock](https://en.wikipedia.org/wiki/Global_interpreter_lock) preventing different Julia threads from doing real computation at the same time.
```

Comment:
How else do you express the alternative to threads really running at the same time?